src/ui/components/ExplorerHome/ExplorerHome.js

```diff
@@ -88,77 +98,164 @@ const useStyles = makeStyles((theme) => ({
+});
+const GRAIN_SORT = deepFreeze({
+  name: Symbol("Grain"),
+  fn: (n) => n.grainEarned,
+});
+const PAGINATION_OPTIONS = deepFreeze([50, 100, 200]);
+
 type ExplorerHomeProps = {|
   +initialView: CredGrainView | null,
+  +currency: CurrencyDetails,
 |};
 
-export const ExplorerHome = ({initialView}: ExplorerHomeProps): ReactNode => {
+export const ExplorerHome = ({
+  initialView,
+  currency: {suffix: currencySuffix, name: currencyName},
+}: ExplorerHomeProps): ReactNode => {
   if (!initialView) return null;
 
   const classes = useStyles();
   const [tab, setTab] = useState<number>(1);
   const [checkboxes, setCheckboxes] = useState({
-    participants: false,
-    organizations: false,
-    bots: false,
-    projects: false,
+    [IdentityTypes.USER]: false,
+    [IdentityTypes.ORGANIZATION]: false,
+    [IdentityTypes.BOT]: false,
+    [IdentityTypes.PROJECT]: false,
+  });
+
+  const allParticipants = useMemo(
+    () => Array.from(initialView.participants()),
+    [initialView.participants()]
+  );
+
+  const tsParticipants = useTableState(allParticipants, {
+    initialRowsPerPage: PAGINATION_OPTIONS[0],
+    initialSort: {
+      sortName: CRED_SORT.name,
+      sortOrder: SortOrders.DESC,
+      sortFn: CRED_SORT.fn,
+    },
   });
-  const {participants, organizations, bots, projects} = checkboxes;
 
-  const handleChange = (event) => {
-    setCheckboxes({...checkboxes, [event.target.name]: event.target.checked});
+  let credTimelineSummary = [];
+  const credAndGrainSummary = {
+    totalCred: 0,
+    totalGrain: 0,
+    avgCred: 0,
+    avgGrain: 0,
   };
 
-  const data = [
-    {title: "Cred This Week", value: 610},
-    {title: "Grain Harvested", value: "6,765g"},
-    {title: "Active Participants", value: 13},
-    {title: "Grain per Cred", value: "22g"},
-  ];
+  // do the summaries of the current page of participants
+  for (const participant of tsParticipants.currentPage) {
+    // add this node's cred to the summary graph
+    credTimelineSummary = participant.credPerInterval.map(
+      (total, i) => (credTimelineSummary[i] || 0) + total
+    );
 
-  const createData = (username, cred, grain, chart) => ({
-    username,
-    cred,
-    grain,
-    chart,
-  });
+    credAndGrainSummary.totalCred += participant.cred;
+    credAndGrainSummary.totalGrain += Number(participant.grainEarned);
+  }
 
-  // sort by cred amount, highest to lowest
-  const nodes = sortBy(initialView.participants(), (n) => -n.cred);
-  // create an array of 0s for the cred summary graph at the top of the page
-  let credTimelineSummary = initialView.intervals().map(() => 0);
+  if (tsParticipants.currentPage.length > 0) {
+    credAndGrainSummary.avgCred =
+      credAndGrainSummary.totalCred / tsParticipants.currentPage.length;
+    credAndGrainSummary.avgGrain =
+      credAndGrainSummary.totalGrain / tsParticipants.currentPage.length;
+  } else {
+    credAndGrainSummary.avgCred = 0;
+    credAndGrainSummary.avgGrain = 0;
+  }
```

Comment:
I'm pretty sure you could put this in a `useEffect` react wrapper that watches `tsParticipants.currentPage` to prevent it from re-aggregating needlessly.

Reply:
I ended up using `useMemo` because these calculations need to be included in the render, and `useMemo` runs before render, whereas `useEffect` runs after render.